### ISSUE TITLE
Only obtain three cell types such as tumor cell, macrophages and neutrophils from bulk data?

### ISSUE THREAD

**Issue by zhangqi234** [ZJUFanLab/bulk2space#8]
Hi, thanks for coming up with this useful tool. 
I have bulk RNAseq data and scRNA-seq data from the same patient which was made by our lab. 
I want to convert bulk transcriptomes into spatially resolved single-cell expression profile. 
Here are my questions:
1.Why do I only obtain three cell types such as tumor cell, macrophages and neutrophils from bulk data? However, there are
    many other celltypes like Fibroblasts, T cell and B cell in my scRNA reference.
2.How to normalize my bulk data?

Thanks,
Qi.

**Comment by SpaTrek** [comment]
Thx, I think there are three reasons,
1. The bulk data quality is not good. The detected number of genes, average expression, and cell-type-specific gene expression would affect the deconvolution result.
2. The reference profile is less heterogeneous, with fewer cell types. Maybe change to another SC reference could improve the result.
3.  Despite the above quality-based problems, I suggest you try more hyperparameters, for example, the top marker gene number. As you can see, the more marker genes may lead to fine clusters because more genes are considered when clustering. In the meantime, this could cause overfitting because not all genes are key determinants for cell identification. In this situation, you may decrease the marker gene number.

In my opinion, Bulk2Space is designed to fit in all kinds of circumstances, the choice of parameters should also be adjusted according to the actual situation.

**Comment by zhangqi234** [comment]
Thanks very much! I try my data again!
